### PR TITLE
Make generic class interface for cohort based filtering

### PR DESCRIPTION
**Pull request by gaugup**
## Description

Introducing the class `FilterDataWithCohortFilters` to provide a generic data filtering using cohorts. This has all the current functionality in cohort_filter.py which are not available under the class `FilterDataWithCohortFilters`. User needs to setup the class using dataset, model, feature types etc and they can then filter the supplied dataset using the function `filter_data_from_cohort`. 

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: erroranalysis/erroranalysis/_internal/cohort_filter.py [hunk 1]
```diff
@@ -26,259 +29,346 @@ def filter_from_cohort(analyzer, filters, composite_filters):
...
-        if true_y[i] == pred_y[i]:
-            if true_y[i] == classes[0]:
-                # True positive == 0
-                classification_outcome.append(0)
+        :param filters: The filters.
+        :type filters: list[dict]
+        :param composite_filters: The composite filters.
+        :type composite_filters: list[dict]
+        :return: The filtered dataset converted to a pandas DataFrame.
+        :param original_data_columns: Whether to just include the original
```
Comment (imatiach-msft) [inline erroranalysis/erroranalysis/_internal/cohort_filter.py:103]: nit: maybe include_original_columns might be a better variable name for a boolean type?  original_data_columns makes me think that I need the list of columns themselves.

File: erroranalysis/erroranalysis/_internal/cohort_filter.py [hunk 2]
```diff
@@ -26,259 +29,346 @@ def filter_from_cohort(analyzer, filters, composite_filters):
...
-    # compute the necessary column data on the fly
-    if model_task == ModelTask.CLASSIFICATION and filters:
-        for filter in filters:
-            if COLUMN in filter:
-                column = filter[COLUMN]
-                if column == CLASSIFICATION_OUTCOME:
-                    return True
-    return False
+class FilterDataWithCohortFilters:
+    def __init__(self, model: Any, dataset: pd.DataFrame,
```
Comment (imatiach-msft) [inline erroranalysis/erroranalysis/_internal/cohort_filter.py:54]: nice refactor!

**Review by imatiach-msft — APPROVED** [review]

File: erroranalysis/erroranalysis/_internal/cohort_filter.py (hunk 1 above)
Comment (imatiach-msft) [reply erroranalysis/erroranalysis/_internal/cohort_filter.py:103]: same above on line 22

Comment (gaugup) [reply erroranalysis/erroranalysis/_internal/cohort_filter.py:103]: Thanks for the suggestion. Changed it to include_original_columns_only.

File: erroranalysis/erroranalysis/_internal/cohort_filter.py (hunk 2 above)
Comment (gaugup) [reply erroranalysis/erroranalysis/_internal/cohort_filter.py:54]: Thanks

**Pull request merged**